src/ModelContextProtocol/Client/McpClient.cs

```diff
@@ -95,13 +118,14 @@ public async Task ConnectAsync(CancellationToken cancellationToken = default)
         try
         {
             // Send initialize request
-            var initializeResponse = await this.SendRequestAsync(
+            Debug.Assert(_options.ClientInfo is not null, "ClientInfo should be set by the constructor");
+                var initializeResponse = await this.SendRequestAsync(
                 RequestMethods.Initialize,
                 new InitializeRequestParams
                 {
                     ProtocolVersion = _options.ProtocolVersion,
                     Capabilities = _options.Capabilities ?? new ClientCapabilities(),
-                    ClientInfo = _options.ClientInfo
+                    ClientInfo = _options.ClientInfo!
```

Comment:
```suggestion
                    ClientInfo = _options.ClientInfo
```

nit: Do you need the null forgiving operator *and* Debug.Assert?

Reply:
> Do you need the null forgiving operator and Debug.Assert?

For netstandard2.0, unfortunately, yes. Debug.Assert in netstandard2.0 doesn't have the annotations.